great_expectations/agent/agent.py

```diff
@@ -0,0 +1,124 @@
+import os
+from dataclasses import dataclass
+from typing import Dict, Optional
+
+from pika.exceptions import AMQPError
+
+from great_expectations import get_context
+from great_expectations.agent.event_handler import ShutdownRequest, EventHandler
+from great_expectations.agent.message_service.rabbit_mq_client import (
+    RabbitMQClient,
+    ClientError,
+)
+from great_expectations.agent.message_service.subscriber import (
+    Subscriber,
+    OnMessageCallback,
+    SubscriberError,
+)
+from great_expectations.agent.models import Event
+from great_expectations.data_context import CloudDataContext
+
+
+HandlerMap = Dict[str, OnMessageCallback]
+
+
+@dataclass(frozen=True)
+class GXAgentConfig:
+    """GXAgent configuration.
+
+    Attributes:
+        organization_id: GX Cloud organization identifier
+        broker_url: address of broker service
+    """
+
+    organization_id: str
+    broker_url: str
```

Comment:
Can we make this a pydantic dataclass and use a special URL type for broker_url

https://docs.pydantic.dev/latest/usage/dataclasses/
https://docs.pydantic.dev/latest/usage/types/#urls
```python
from pydantic.dataclasses import dataclass
from pydantic import HttpUrl

```


```suggestion
@dataclass(frozen=True)
class GXAgentConfig:
    """GXAgent configuration.

    Attributes:
        organization_id: GX Cloud organization identifier
        broker_url: address of broker service
    """

    organization_id: str
    broker_url: HttpUrl
```

Reply:
unfortunately not -- these will be `amqps` protocol, not `http/https`

Reply:
actually we can! just need to use the `AmqpDsn` type instead

Reply:
You can use `AnyUrl`

Reply:
Oh nice! Didn't notice the `AmqpDsn ` type, even better!